code/modules/mob/living/living.dm

```diff
@@ -937,9 +953,191 @@ Sorry Giacom. Please don't be mad :(
 	else
 		src << "<span class='warning'>You don't have the dexterity to do this!</span>"
 	return
-
 /mob/living/proc/can_use_guns(var/obj/item/weapon/gun/G)
 	if (!IsAdvancedToolUser())
 		src << "<span class='warning'>You don't have the dexterity to do this!</span>"
 		return 0
-	return 1
\ No newline at end of file
+	return 1
+
+/mob/living/carbon/proc/update_stamina()
+	return
+
+/mob/living/carbon/human/update_stamina()
+	if(staminaloss)
+		var/total_health = (health - staminaloss)
+		if(total_health <= config.health_threshold_crit && !stat)
+			src << "<span class='notice'>You're too exhausted to keep going...</span>"
+			Weaken(5)
+		update_health_hud()
+
+/mob/proc/update_sight()

```

Comment:
The proc to handle changes to the mob vars "sight", "see_invisible", and "see_in_dark".